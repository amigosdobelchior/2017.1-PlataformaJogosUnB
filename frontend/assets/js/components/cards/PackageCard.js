import _ from 'lodash'
import React from "react"
import PropTypes from 'prop-types'
import {Card, Button, Grid, Icon, Modal, Message} from "semantic-ui-react"
import ModalPackageCard from "./ModalPackageCard"


export default class PackageCard extends React.Component {

    constructor(props){
        super(props)
        this.state = {
            modalOpen: false
        }
        this.getHandleOpen = this.getHandleOpen.bind(this)
        this.getHandleClose = this.getHandleClose.bind(this)
    }

    reduceKernelPlatforms(packages) {
        let platforms = [];
        if (packages !== undefined) {
            platforms = _.reduce(packages, (platform, bpackage) => {
                const platform_kernel = _.map(bpackage.platforms, (platform_param) => platform_param.kernel);
                return platform.concat(platform_kernel);
            }, []);
        }
        return (_.uniq(platforms));
    }

    getIcon(platform_icon){
        if (platform_icon==='OSX') {
            platform_icon = 'apple'
        }
        return platform_icon.toLowerCase()
    }

    getHandleOpen(){
        this.setState({modalOpen: true})
    }

    getHandleClose(){
        this.setState({modalOpen: false})
    }

    getLicensePlatformModal(value,index){
        const messageLicense = "Ao realizar o download do instalador do jogo. O(A) senhor(a) concorda que a plataforma de jogos da Universidade de Brasília (UnB games), e a Universidade de Brasília não responsabiliza-se por possíveis danos em seu computador nem mal funcionamento dos jogos baixados."
        return (
            <Modal key={'licenseSoftware'} open={this.state.modalOpen} onClose={this.getHandleClose}
                    trigger={
                        <Button basic color='green' onClick={this.getHandleOpen}>
                            <Icon name={this.getIcon(value)} />
                        </Button>
                    }
            >
                <Modal.Header>Termo de uso <Icon color='yellow' name='warning sign'/></Modal.Header>
                <Modal.Content>
                   <Message info>
                        <p>{messageLicense}</p>
                    </Message>
                </Modal.Content>
                <Modal.Actions>
                    <Button color='red' onClick={this.getHandleClose}>
                        <Icon name='remove' />Não aceitar
                    </Button>
                    {this.getPackagesModal(value,index)}
                </Modal.Actions>
            </Modal>
        )
    }

    getPackagesModal(value,index){
        const game_pk = this.props.game_pk

        return (
                <ModalPackageCard key={index}
                    button={
                        <Button color='green'>
                            <Icon name='checkmark' />Aceitar
                        </Button>
                    }
                    game_pk={game_pk}
                    kernel={value}
                    gameName={this.props.gameName}
                    downloads={this.props.downloads}
                />
        )
    }

    getButtonsPlatforms(){
        const buttons_platforms = (this.reduceKernelPlatforms(this.props.packages)).map((value,index)=>
            this.getLicensePlatformModal(value,index)
        )
        if (buttons_platforms.length > 0) {
            return buttons_platforms;
        } else {
            return <Button basic color='red'>Não há instaladores cadastrados</Button>
        }
    }

    render () {

        return (
            <Card fluid>
                <Card.Content header="Download"/>
                <Card.Content>
                    <Grid centered size='large'>
                        <Button.Group>
                            {this.getButtonsPlatforms()}
                        </Button.Group>
                    </Grid>
                </Card.Content>
                <Card.Content extra/>
            </Card>
        );
    }
}

PackageCard.propTypes = {
    downloads: PropTypes.number.isRelated,
    game_pk: PropTypes.number.isRequired,
    packages: PropTypes.array.isRequired,
    gameName: PropTypes.string.isRequired
}
