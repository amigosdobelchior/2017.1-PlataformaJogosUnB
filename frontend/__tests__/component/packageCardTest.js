import React from 'react';
import PackageCard from '../../assets/js/components/cards/PackageCard';
import {Card, Button, Grid} from "semantic-ui-react";
import renderer from 'react-test-renderer';

test('Test render PackageCard', () => {
  const component = renderer.create(
      <Card fluid>
          <Card.Content header="Download"/>
          <Card.Content>
              <Grid centered size='large'>
                  <Button.Group>
                     ['button1','button2','button3']
                  </Button.Group>
              </Grid>
          </Card.Content>
          <Card.Content extra/>
      </Card>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
