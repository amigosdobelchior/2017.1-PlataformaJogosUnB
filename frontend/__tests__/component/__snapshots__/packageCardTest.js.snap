// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render PackageCard 1`] = `
<div
  className="ui fluid card"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Download
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="ui centered grid"
      size="large"
    >
      <div
        className="ui buttons"
      >
        [
        button1
        ,
        button2
        ,
        button3
        ]
      </div>
    </div>
  </div>
  <div
    className="extra content"
  />
</div>
`;
